server/server/src/security.ts

```diff
@@ -18,27 +19,39 @@ import { component, Component, PlatformError, Severity, Status, StatusCode } fro
 
 export class SecurityModel extends TxProcessor {
   private readonly hierarchy: Hierarchy
+  private readonly model: ModelDb
   private readonly allowedSpaces: Map<Ref<Account>, Set<Ref<Space>>> = new Map<Ref<Account>, Set<Ref<Space>>>()
   private readonly publicSpaces: Set<Ref<Space>> = new Set<Ref<Space>>()
 
-  constructor (hierarchy: Hierarchy) {
+  constructor (hierarchy: Hierarchy, model: ModelDb) {
     super()
     this.hierarchy = hierarchy
+    this.model = model
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    model.findAll(core.class.Space, {}).then(spaces => {
```

Comment:
Not a good style, could you please create a static async create function and do all actions inside it. So it will be possible to properly test and be same all is initialised before it will be used.